### ISSUE TITLE
TypeError: F is not a function when using @emotion/styled

### ISSUE THREAD

**Issue by fabien-thebaud-ariadnext** [vitest-dev/vitest#3350]
### Describe the bug

My team has built a library  (called _ui-react_) of React components styled with [Emotion](https://emotion.sh/docs/@emotion/styled). Both react and emotion are peer dependencies.
Since introducing these components to our main project, most of our tests are now failing (coverage down from 90 to 20%), with just this error message:
```
TypeError: F is not a function
 ❯ node_modules/@mycompany/ui-react/dist/@mycompany/ui-react.js:18201:12
```

The issue is the same whether we are using the `styled('div')({})` or the `styled.div({})` syntax. The error message changes slightly:

```
TypeError: F.div is not a function
 ❯ node_modules/@mycompany/ui-react/dist/@mycompany/ui-react.js:18201:14
```

If I look at the _ui-react_ bundle:
```
import b1 from "react";
import F from "@emotion/styled";
```
We are indeed importing the default "styled" function from emotion. Its type is [CreateStyle](https://github.com/emotion-js/emotion/blob/main/packages/styled/types/index.d.ts), which extends [BaseCreateStyle](https://github.com/emotion-js/emotion/blob/main/packages/styled/types/base.d.ts#L118), which is a function:

```
/**
 * @desc
 * This function accepts a React component or tag ('div', 'a' etc).
 *
 * @example styled(MyComponent)({ width: 100 })
 * @example styled(MyComponent)(myComponentProps => ({ width: myComponentProps.width })
 * @example styled('div')({ width: 100 })
 * @example styled('div')<Props>(props => ({ width: props.width })
 */
export interface CreateStyled {
    ...
    <several function signatures are defined here>
    ...
}
```

So I'm not sure what to do here, the rest of the project if working fine, only vitest is complaining about the typing.
Could this be solved with the right vitest configuration or is it a bug?

### Reproduction

See instructions in the README file
https://github.com/fthebaud/repro-vitest-styled-issue

### System Info

```shell
System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (12) x64 Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
    Memory: 2.98 GB / 15.30 GB
    Container: Yes
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 18.14.1 - ~/.nvm/versions/node/v18.14.1/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 9.3.1 - ~/.nvm/versions/node/v18.14.1/bin/npm
  Browsers:
    Chrome: 113.0.5672.92
    Firefox: 102.10.0esr
  npmPackages:
    @vitejs/plugin-react: 4.0.0 => 4.0.0 
    @vitest/coverage-istanbul: 0.30.1 => 0.30.1 
    vite: 4.3.1 => 4.3.1 
    vitest: 0.30.1 => 0.30.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by fabien-thebaud-ariadnext** [comment]
Here's a repo that reproduce the issue. Instructions are in the README.
https://github.com/fthebaud/repro-vitest-styled-issue

**Comment by sheremet-va** [comment]
Some libraries bundle their code for bundlers to consume so they don't follow Node.js ESM semantics. In this case `emotion` has a default export that is not actually a function when imported inside Node.js, but an object with another "default" property. This was fixed in [@emotion/styled@11.11.0](https://github.com/emotion-js/emotion/releases/tag/%40emotion%2Fstyled%4011.11.0)

When updating the package I get a different error:

```
TypeError: Cannot read properties of null (reading 'useContext')
```

Why this happens I do not know, it's probably imported somewhere from ESM and somewhere from CJS, so it creates two different react instances.

**Comment by fthebaud** [comment]
I've upgraded emotion in my original project and it fixed the issue, I didn't have to add anything to `deps.inline`.
Thanks for your help @sheremet-va, much appreciated.
